syntax = "proto3";

package fbsrankings.messages.query;

import "fbsrankings/messages/options/options.proto";

message TeamRecordValueBySeasonWeekResult {
    string team_id = 1;
    string name = 2;
    int32 wins = 3;
    int32 losses = 4;
}

message TeamRecordBySeasonWeekValue {
    string record_id = 1;
    string season_id = 2;
    int32 year = 3;
    optional int32 week = 4;
    repeated TeamRecordValueBySeasonWeekResult values = 5;
}

message TeamRecordBySeasonWeekResult {
    string query_id = 1;
    optional TeamRecordBySeasonWeekValue record = 2;
}

message TeamRecordBySeasonWeekQuery {
    string query_id = 1;
    string season_id = 2;
    optional int32 week = 3;

    option (fbsrankings.messages.options.topic) = "fbsrankings.query.team_record_by_season_week";
}
